sonar-text-plugin/src/test/java/org/sonar/plugins/text/core/CommonSensorTest.java

```diff
@@ -0,0 +1,208 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.text.core;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.rule.Checks;
+import org.sonar.api.batch.rule.internal.ActiveRulesBuilder;
+import org.sonar.api.batch.rule.internal.NewActiveRule;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.error.AnalysisError;
+import org.sonar.api.batch.sensor.internal.DefaultSensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.batch.sensor.issue.Issue;
+import org.sonar.api.batch.sensor.issue.IssueLocation;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.LogTesterJUnit5;
+import org.sonar.plugins.text.CommonPlugin;
+import org.sonar.plugins.text.api.CommonCheck;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+class CommonSensorTest {
+
+  @RegisterExtension
+  public LogTesterJUnit5 logTester = new LogTesterJUnit5();
+
+  private static final String REPOSITORY_KEY = CommonPlugin.REPOSITORY_KEY;
+
+  @TempDir
+  protected File baseDir;
+  protected SensorContextTester context;
+
+  @BeforeEach
+  void setup() {
+    context = SensorContextTester.create(baseDir);
+    MapSettings settings = new MapSettings();
+    context.setSettings(settings);
+  }
+
+  @Test
+  void should_return_sensor_descriptor() {
+    DefaultSensorDescriptor descriptor = new DefaultSensorDescriptor();
+    sensor().describe(descriptor);
+    assertThat(descriptor.name()).isEqualTo("Common Sensor");
+    assertThat(descriptor.languages()).isEmpty();
+  }
+
+  @Test
+  void empty_file_should_raise_no_issue() {
+    // TODO replace by rules supported by this plugin
+    analyse(sensor("S2260"), inputFile("empty.ts", ""));
+    assertThat(context.allIssues()).as("No issue must be raised").isEmpty();
+  }
+
+  @Test
+  void stop_on_cancellation() {
+    context.setCancelled(true);
```

Comment:
This test does also pass when we set `context.setCancelled(false);`. Something is wrong here.